src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java

```diff
@@ -307,13 +324,17 @@ private boolean updateImpl(String id, Function<String, T> getter, Function<Optio
      * @return List of elements or empty if none match
      */
     public List<T> scatterGather(DetachedCriteria criteria) {
-        return daos.stream().map(dao -> {
+        List<T> results = new ArrayList<>();
```

Comment:
Why this change ?

Reply:
as it's scatter gather, need to pass shard id (`i` in this case) in the metric name.